qa-tests-backend/src/test/groovy/DeclarativeConfigTest.groovy

```diff
@@ -0,0 +1,586 @@
+        }
+
+        // No auth provider should be created.
+        assert AuthProviderService.getAuthProviderService().
+                getAuthProviders(
+                        AuthproviderService.GetAuthProvidersRequest.newBuilder()
+                                .setName(VALID_AUTH_PROVIDER.getName()).build()
+                )
+                .getAuthProvidersCount() == 0
+
+        when:
+        orchestrator.deleteConfigMap(CONFIGMAP_NAME, DEFAULT_NAMESPACE)
+
+        then:
+        // Only the config map health status should exist, all others should be removed.
+        withRetry(5, 60) {
+            def response = IntegrationHealthService.getDeclarativeConfigHealthInfo()
+            assert response.getIntegrationHealthCount() == 1
+            def configMapHealth = response.getIntegrationHealth(0)
+            assert configMapHealth
+            assert configMapHealth.getName().contains("Config Map")
+            assert configMapHealth.getErrorMessage() == ""
+            assert configMapHealth.getStatus() == Status.HEALTHY
+        }
+    }
+
+    // Helpers
+
+    // createDefaultSetOfResources creates the following resources:
+    //  - permission set with valid configuration.
+    //  - access scope with valid configuration.
+    //  - role with valid configuration, referencing the previously created permission set / access scope.
+    //  - auth provider with valid configuration, and two groups (one default, one separate)
+    private createDefaultSetOfResources(String configMapName, String namespace) {
+        orchestrator.createConfigMap(configMapName,
+                [
+                        (PERMISSION_SET_KEY): VALID_PERMISSION_SET_YAML,
+                        (ACCESS_SCOPE_KEY): VALID_ACCESS_SCOPE_YAML,
+                        (ROLE_KEY): VALID_ROLE_YAML,
+                        (AUTH_PROVIDER_KEY): VALID_AUTH_PROVIDER_YAML,
+                ], namespace)
+    }
+
+    // updateConfigMapValue updates a key / value pair within the given config map.
+    private updateConfigMapValue(String configMapName, String namespace, String key, String value) {
+        def configMap = orchestrator.getConfigMap(configMapName, namespace)
+        configMap.data.put(key, value)
+        orchestrator.createConfigMap(configMap)
+    }
+
+    // verifyDeclarativeRole will verify that the expected role exists within the API and shares the same values.
+    // The retrieved role from the API will be returned.
+    private Role verifyDeclarativeRole(Role expectedRole, String permissionSetID, String accessScopeID) {
+        def role = RoleService.getRole(expectedRole.getName())
+        assert role : "declarative role ${expectedRole.getName()} does not exist"
+        assert role.getName() == expectedRole.getName()
+        assert role.getDescription() == expectedRole.getDescription()
+        assert role.getTraits().getOrigin() == expectedRole.getTraits().getOrigin()
```

Comment:
I guess I keep it for now, at the very least we only have to change it for the expected role in case one of our naming discussion has the outcome to rename this one 😛